### PR TITLE
add controller to get multiple filter by id

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/main/resources/application-local.yaml [hunk 1]
```diff
@@ -8,3 +8,10 @@ spring:
 powsybl-ws:
   database:
     host: localhost
+
```
Comment (SlimaneAmar) [inline src/main/resources/application-local.yaml:11]: Why these properties ?

File: src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java [hunk 2]
```diff
@@ -833,6 +838,158 @@ public void testIdentifierListFilter() throws Exception {
...
+                mvc.perform(get("/" + FilterApi.API_VERSION + "/filters/metadata").params(params)
+                                .contentType(APPLICATION_JSON))
+                        .andExpect(status().isOk())
+                        .andReturn().getResponse().getContentAsString(),
+                new TypeReference<>() {
+                });
+        Date dateModification = filterAttributes.get(0).getModificationDate();
+        Date dateModification2 = filterAttributes.get(1).getModificationDate();
+
+        var expected = "[ {\n" +
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java:880]: Oh !
No hardcoded json string
Rather use DTO.builder and objectMapper.writeValueAsString

File: src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java [hunk 3]
```diff
@@ -833,6 +838,158 @@ public void testIdentifierListFilter() throws Exception {
...
+
+        List<FilterEquipments> filterEquipments = objectMapper.readValue(
+                mvc.perform(get("/" + FilterApi.API_VERSION + "/filters/export").params(params)
+                                .contentType(APPLICATION_JSON))
+                        .andExpect(status().isOk())
+                        .andReturn().getResponse().getContentAsString(),
+                new TypeReference<>() {
+                });
+
+        var expected = "[ {\n" +
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java:972]: Same

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/resources/application-local.yaml (hunk 1 above)
Comment (bendaamerahm) [reply src/main/resources/application-local.yaml:11]: my bad i forgot to delete them

File: src/main/java/org/gridsuite/filter/server/FilterController.java [hunk 4]
```diff
@@ -135,4 +138,15 @@ public ResponseEntity<List<IdentifiableAttributes>> exportFilter(@PathVariable("
             .body(identifiables))
             .orElse(ResponseEntity.notFound().build());
     }
+
+    @GetMapping(value = "/filters/export", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Export list of filters to JSON format")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The filters on JSON format")})
+    public ResponseEntity<List<FilterEquipments>> exportFilters(@RequestParam("ids") List<UUID> ids,
```
[the remark this answers is not in the document]
Comment (bendaamerahm) [reply src/main/java/org/gridsuite/filter/server/FilterController.java:145]: we gonna create another PR for this modification.

**Review by FranckLecuyer — APPROVED** [review]

File: src/main/java/org/gridsuite/filter/server/FilterService.java [hunk 5]
```diff
@@ -627,4 +637,33 @@ public Optional<List<IdentifiableAttributes>> exportFilter(UUID id, UUID network
...
     }
+
+    public List<FilterEquipments> exportFilters(List<UUID> ids, UUID networkUuid, String variantId) {
+        return getFilters(ids).stream()
+                .map(filter -> getFilterEquipments(filter, getIdentifiableAttributes(filter, networkUuid, variantId)))
+                .collect(Collectors.toList());
+    }
+
+    private FilterEquipments getFilterEquipments(AbstractFilter filter, List<IdentifiableAttributes> identifiableAttributes) {
+        if (filter.getType() == FilterType.IDENTIFIER_LIST) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/filter/server/FilterService.java:648]: No
Add an abstract method in the `AbstractFilter` class

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**